### PR TITLE
Better Sprite catalogue

### PR DESCRIPTION
**Pull request by Bod9001**
the index is generated in runtime instead of in the editor, hopefully resolves a few Issues with the catalogue

### CODE REVIEW

File: UnityProject/AddressablePackingProjects/SoundAndMusic/Packages/packages-lock.json [hunk 1]
```diff
@@ -61,7 +61,7 @@
       "url": "https://packages.unity.com"
     },
     "com.unity.scriptablebuildpipeline": {
-      "version": "1.15.2",
```
Comment (NoooneyDude) [inline UnityProject/AddressablePackingProjects/SoundAndMusic/Packages/packages-lock.json]: Hmm, this wasn't available in 2021.1.14f1 for me. Probably don't want this for now? Would it cause issues if anyone is still on 14f1?

Comment (Bod9001) [reply UnityProject/AddressablePackingProjects/SoundAndMusic/Packages/packages-lock.json]: ok, yeah, the forced Serialisation changes drowned a lot of stuff

**Review by NoooneyDude — DISMISSED** [review]

**Comment by corp-0** [comment]
test never finished

**Comment by corp-0** [comment]
how can I even read the code changes

**Comment by NoooneyDude** [comment]
> how can I even read the code changes



Though it does take a few minutes for the browser to respond...

**Comment by Bod9001** [comment]
I will run tests locally

**Review by NoooneyDude — APPROVED** [review]
Looks like it's just the labeler failing.

**Review by corp-0 — APPROVED** [review]

**Pull request merged**